app/agents/voice/automatic/services/mcp/automatic_client.py

```
@@ -46,50 +115,171 @@ async def post(self, method: str, params: Optional[Dict[str, Any]] = None) -> Di
+                                    converted_response = {
+                                        "jsonrpc": "2.0",
+                                        "id": 1,
+                                        "result": {"tools": tools_array}
+                                    }
+                                    logger.info(f"Converted non-event response to JSON-RPC format with {len(tools_array)} tools")
+                                    return converted_response
+                                
+                        except json.JSONDecodeError as e:
                             logger.error(f"Failed to decode JSON from stream: {json_str}")
+                            logger.error(f"JSON decode error: {e}")
                             raise ValueError("Received malformed JSON from server.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)